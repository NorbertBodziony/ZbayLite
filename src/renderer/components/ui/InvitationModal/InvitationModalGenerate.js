import React from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'

import Grid from '@material-ui/core/Grid'
import { withStyles } from '@material-ui/core/styles'
import { Typography, TextField, FormControlLabel, Checkbox } from '@material-ui/core'
import LoadingButton from '../LoadingButton'
import InvitationModal from './InvitationModal'

const styles = theme => ({
  warrning: {
    marginTop: theme.spacing(1)
  },
  warrningText: {
    maxWidth: 400
  },
  inputDiv: {
    width: 45,
    '& input': {
      textAlign: 'center',
      fontSize: '1.25rem',
      fontWeight: 500,
      paddingTop: 4,
      paddingBottom: 3,
      paddingLeft: 5,
      paddingRight: 0
    }
  },
  amountDiv: {
    marginTop: theme.spacing(1)
  },
  zecAmount: {
    marginTop: theme.spacing(1)
  },
  checkboxDiv: {
    maxWidth: 450,
    marginTop: theme.spacing(1),
    '& span': {
      fontSize: '0.8rem'
    }
  },
  buttonDiv: {
    marginTop: theme.spacing(2)
  },
  button: {
    height: 60,
    width: 350,
    fontSize: '0.9rem',
    backgroundColor: theme.palette.colors.zbayBlue
  },
  checkboxRoot: {
    color: theme.palette.colors.zbayBlue,
    '&$checked': {
      color: theme.palette.colors.zbayBlue
    }
  },
  checked: {}
})

export const InvitationModalGenerate = ({
  classes,
  open,
  handleClose,
  zecRate,
  amount,
  includeAffiliate,
  affiliate,
  generateInvitation,
  setAmount,
  setStep,
  isLoading,
  setLoading
}) => (
  <InvitationModal
    open={open}
    handleClose={handleClose}
    title={`Invite a friend`}
    info={`Invite friends to Zbay and (optionally)
     give them
  some funds to get started`}
  >
    <Grid item className={classes.warrning}>
      <Typography variant='body2' className={classes.warrningText}>
        ( This will generate a special invite link you can share with a friend. Once your friend
        installs Zbay they can open it to reclaim funds. You can use it to reclaim the funds
        yourself too - just dont't lose the link! )
      </Typography>
    </Grid>
    <Grid container justify='center' align='center' className={classes.amountDiv}>
      <Grid item>
        <Typography variant='h6'>$ </Typography>
      </Grid>
      <Grid item>
        <TextField
          classes={{
            input: classes.input
          }}
          className={classes.inputDiv}
          type='number'
          value={amount}
          InputProps={{ inputProps: { min: 0, max: 99 } }}
          onChange={e => setAmount(e.target.value)}
        />
      </Grid>
      <Grid item>
        <Typography variant='h6'> USD</Typography>
      </Grid>
    </Grid>
    <Grid item className={classes.zecAmount}>
      <Typography variant='body2'>{(amount / zecRate).toFixed(4)} ZEC</Typography>
    </Grid>
    <Grid item>
      <FormControlLabel
        control={
          <Checkbox
            checked={affiliate}
            onChange={e => {
              includeAffiliate(e.target.checked)
            }}
            classes={{ root: classes.checkboxRoot, checked: classes.checked }}
          />
        }
        className={classes.checkboxDiv}
        label={`Include your affiliate code, so that you'll earn 1% of their purchases. (They can disable this at any time.)`}
      />
    </Grid>
    <Grid item className={classes.buttonDiv}>
      <LoadingButton
        variant='contained'
        size='large'
        color='primary'
        type='submit'
        fullWidth
        inProgress={isLoading}
        className={classes.button}
        onClick={async () => {
          setLoading(true)
          await generateInvitation(1)
          setStep(1)
          setLoading(false)
        }}
      >
        Generate invitation
      </LoadingButton>
    </Grid>
  </InvitationModal>
)

InvitationModalGenerate.propTypes = {
  classes: PropTypes.object.isRequired,
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  zecRate: PropTypes.number.isRequired,
  amount: PropTypes.number.isRequired,
  includeAffiliate: PropTypes.func.isRequired,
  setStep: PropTypes.func.isRequired,
  affiliate: PropTypes.bool.isRequired,
  setAmount: PropTypes.func.isRequired,
  generateInvitation: PropTypes.func.isRequired,
  setLoading: PropTypes.func.isRequired,
  isLoading: PropTypes.func.isRequired
}

export default R.compose(
  React.memo,
  withStyles(styles)
)(InvitationModalGenerate)
