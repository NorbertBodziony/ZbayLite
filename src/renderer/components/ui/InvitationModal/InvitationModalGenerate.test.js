/* eslint import/first: 0 */
import React from 'react'
import { shallow } from 'enzyme'

import { mockClasses } from '../../../../shared/testing/mocks'
import { InvitationModalGenerate } from './InvitationModalGenerate'

describe('InvitationModalGenerate', () => {
  it('renders component', () => {
    const result = shallow(
      <InvitationModalGenerate
        classes={mockClasses}
        open
        zecRate={1}
        amount={9}
        handleClose={jest.fn()}
        setStep={jest.fn()}
        includeAffiliate={jest.fn()}
        setAmount={jest.fn()}
        affiliate
      />
    )
    expect(result).toMatchSnapshot()
  })
})
