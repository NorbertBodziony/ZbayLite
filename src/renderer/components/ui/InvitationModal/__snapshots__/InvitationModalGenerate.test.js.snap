// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvitationModalGenerate renders component 1`] = `
<Memo(WithStyles(InvitationModal))
  info="Invite friends to Zbay and (optionally)
     give them some funds to get started"
  title="Invite a friend"
>
  <WithStyles(ForwardRef(Grid))
    className="warrning-mock"
    item={true}
  >
    <WithStyles(ForwardRef(Typography))
      variant="body2"
    >
      This will generate a special invite link you can share with a friend. Once your friend installs Zbay they can open it to reclaim funds. You can use it to reclaim the funds yourself too - just dont't lose the link!
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Typography))
      className="fieldTitle-mock"
      variant="subtitle2"
    >
      Funds to give to friend
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="divMoney-mock"
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      className="moneyDiv-mock"
      item={true}
      xs={true}
    >
      <WithStyles(ForwardRef(TextField))
        InputProps={
          Object {
            "className": "moneyInput-mock",
            "endAdornment": <ForwardRef(WithStyles)
              position="end"
            >
              <span
                className="inputMark-mock"
              >
                USD
              </span>
            </ForwardRef(WithStyles)>,
            "inputProps": Object {
              "max": 99,
              "min": 0,
            },
          }
        }
        name="usd"
        onChange={[Function]}
        placeholder="0.00"
        value="0.00"
        variant="outlined"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      className="exchangeDiv-mock"
      container={true}
      item={true}
      justify="center"
    >
      <Memo(Icon)
        src="test-file-stub"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="moneyDiv-mock"
      item={true}
      xs={true}
    >
      <WithStyles(ForwardRef(TextField))
        InputProps={
          Object {
            "className": "moneyInput-mock",
            "endAdornment": <ForwardRef(WithStyles)
              position="end"
            >
              <span
                className="inputMark-mock"
              >
                ZEC
              </span>
            </ForwardRef(WithStyles)>,
          }
        }
        name="zec"
        onChange={[Function]}
        placeholder="0.00"
        value="9.0000"
        variant="outlined"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(FormControlLabel))
      className="checkboxDiv-mock"
      control={
        <ForwardRef(WithStyles)
          checked={true}
          checkedIcon={
            <Memo
              style={
                Object {
                  "fontSize": 18,
                }
              }
            />
          }
          className="checkbox-mock"
          color="primary"
          icon={
            <Memo
              style={
                Object {
                  "fontSize": 18,
                }
              }
            />
          }
          onChange={[Function]}
        />
      }
      label={
        <ForwardRef(WithStyles)
          className="checkboxLabel-mock"
          variant="body2"
        >
          Include your affiliate code, so that you'll earn 1% of their purchases. (They can disable this at any time.)
        </ForwardRef(WithStyles)>
      }
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="buttonDiv-mock"
    item={true}
  >
    <WithStyles(LoadingButton)
      className="button-mock"
      color="primary"
      disabled={false}
      fullWidth={true}
      inProgress={[MockFunction]}
      onClick={[Function]}
      size="large"
      type="submit"
      variant="contained"
    >
      Generate invitation
    </WithStyles(LoadingButton)>
  </WithStyles(ForwardRef(Grid))>
</Memo(WithStyles(InvitationModal))>
`;
